base/math.jl

```diff
@@ -41,9 +41,11 @@ end
 
 """
     clamp(x, lo, hi)
+    clamp(x, T)
```

Comment:
Typically in Base APIs, the type argument comes first. So this should be `clamp(T, x)`.